include/savedata_misc.h

```diff
@@ -0,0 +1,58 @@
+#ifndef POKEPLATINUM_SAVEDATA_MISC_H
+#define POKEPLATINUM_SAVEDATA_MISC_H
+
+#include "struct_defs/sentence.h"
+#include "strbuf.h"
+#include "struct_decls/struct_02027854_decl.h"
+#include "struct_decls/struct_02027860_decl.h"
+#include "struct_defs/struct_02027854.h"
+#include "struct_defs/struct_02027860.h"
+#include "savedata.h"
+#include "constants/savedata/save_table.h"
+#include "constants/savedata/savedata_misc.h"
+
+typedef struct ExtraSaveKey {
+    u32 keys[EXTRA_SAVE_TABLE_ENTRY_MAX - 1];
+    u32 oldKeys[EXTRA_SAVE_TABLE_ENTRY_MAX - 1];
+    u8 keyFlags[EXTRA_SAVE_TABLE_ENTRY_MAX - 1];
+    u8 filler[3];
+} ExtraSaveKey;
+
+typedef struct MiscSaveBlock {
+    UnkStruct_02027854 unk_00[128];
+    UnkStruct_02027860 unk_680;
+    u16 rivalName[RIVAL_NAME_LEN + 1];
+    u16 tabletName[TABLET_NAME_LEN + 1]; // used in shaymin event
+    u16 favoriteMon;
+    u8 favoriteMonForm:7;
+    u8 favoriteMonIsEgg:1;
+    u8 extraSaveInitalized:1;
+    u8 vsRecorderColor:4;
+    u8 filler:3;
```

Comment:
Spaces on both sides of the `:` for bitflags:

```suggestion
    u8 favoriteMonForm : 7;
    u8 favoriteMonIsEgg : 1;
    u8 extraSaveInitalized : 1;
    u8 vsRecorderColor : 4;
    u8 filler : 3;
```